packages/aws-cdk-lib/core/lib/asset-staging.ts

```diff
@@ -452,17 +452,43 @@ export class AssetStaging extends Construct {
           sourcePath: this.sourcePath,
           bundleDir,
           ...options,
+          securityOpt: options.securityOpt ?? '',
+          user: options.user ?? this.determineUser(),
+          volumes: [...(options.volumes ?? [])],
+          workingDirectory:
+            options.workingDirectory ?? AssetStaging.BUNDLING_INPUT_DIR,
         };
 
+        // Add the asset input and output volumes based on BundlingFileAccess setting
         switch (options.bundlingFileAccess) {
           case BundlingFileAccess.VOLUME_COPY:
-            new AssetBundlingVolumeCopy(assetStagingOptions).run();
```

Comment:
Actually I prefer these individual helper classes over the proposed `DockerVolumeHelper` - even at the expense of some duplication. Since volume commands differ so much between the two options, a common helper class makes it harder to reason through.